### PR TITLE
Add Flag for SSZ Encoding

### PR DESCRIPTION
**Pull request by nisdas**
Part of #3147 
- [x] Add flag for ssz encoding
- [x] Encode based on the flag's value

### CODE REVIEW

File: shared/cmd/flags.go [hunk 1]
```diff
@@ -107,6 +107,12 @@ var (
 			"would whitelist connections to peers on your local network only. The default " +
 			"is to accept all connections.",
 	}
+	// P2PEncoding defines the encoding format for p2p messages.
+	P2PEncoding = cli.IntFlag{
+		Name:  "p2p-encoding",
+		Usage: "The encoding format of messages sent over the wire. The default is 0, which represents ssz",
+		Value: 0,
+	}
```
Comment (shayzluf) [inline shared/cmd/flags.go]: can we consider changing it from int to boolean something like ssz-compression ?

Comment (nisdas) [reply shared/cmd/flags.go]: @shayzluf we ideally want to support a number of encoding formats, ex: protobufs,json etc. This isnt solely for specifying snappy compression, so this is why I have the flag as an int

Comment (shayzluf) [reply shared/cmd/flags.go]: sg

File: shared/cmd/flags.go [hunk 2]
```diff
@@ -107,6 +107,12 @@ var (
 			"would whitelist connections to peers on your local network only. The default " +
 			"is to accept all connections.",
 	}
+	// P2PEncoding defines the encoding format for p2p messages.
+	P2PEncoding = cli.IntFlag{
```
Comment (prestonvanloon) [inline shared/cmd/flags.go]: Please use string flag

File: beacon-chain/p2p/service.go [hunk 3]
```diff
@@ -105,7 +105,13 @@ func (s *Service) Status() error {
 
 // Encoding returns the configured networking encoding.
 func (s *Service) Encoding() encoder.NetworkEncoding {
-	// TODO(3147): Return based on flag value
+	encoding := s.cfg.Encoding
+	switch encoding {
```
Comment (prestonvanloon) [inline beacon-chain/p2p/service.go]: Please add default to panic if invalid flag.

Comment (nisdas) [reply beacon-chain/p2p/service.go]: alright,done

**Pull request merged**